plugins/outputs/warp10/warp10.go

```diff
@@ -0,0 +1,179 @@
+package warp10
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+type Warp10 struct {
+	Prefix string
+
+	WarpUrl string
+
+	Token string
+
+	Debug bool
+}
+
+var sampleConfig = `
+  # prefix for metrics class Name
+  prefix = "Prefix"
+  ## POST HTTP(or HTTPS) ##
+  # Url name of the Warp 10 server
+  warp_url = "WarpUrl"
+  # Token to access your app on warp 10
+  token = "Token"
+  # Debug true - Prints Warp communication
+  debug = false
+`
+
+type MetricLine struct {
+	Metric    string
+	Timestamp int64
+	Value     string
+	Tags      string
+}
+
+func (o *Warp10) Connect() error {
+	return nil
+}
+
+func (o *Warp10) Write(metrics []telegraf.Metric) error {
+
+	var out io.Writer = ioutil.Discard
+	if o.Debug {
+		out = os.Stdout
+	}
+
+	if len(metrics) == 0 {
+		return nil
+	}
+	var now = time.Now()
+	collectString := make([]string, 0)
+	index := 0
+	for _, mm := range metrics {
+
+		for k, v := range mm.Fields() {
+
+			metric := &MetricLine{
+				Metric:    fmt.Sprintf("%s%s", o.Prefix, mm.Name()+"."+k),
+				Timestamp: now.Unix() * 1000000,
+			}
+
+			metricValue, err := buildValue(v)
+			if err != nil {
+				log.Printf("Warp: %s\n", err.Error())
+				continue
+			}
+			metric.Value = metricValue
+
+			tagsSlice := buildTags(mm.Tags())
+			metric.Tags = strings.Join(tagsSlice, ",")
+
+			messageLine := fmt.Sprintf("%d// %s{%s} %s\n", metric.Timestamp, metric.Metric, metric.Tags, metric.Value)
+
+			collectString = append(collectString, messageLine)
+			index += 1
+		}
+	}
+	payload := fmt.Sprint(strings.Join(collectString, "\n"))
+	//defer connection.Close()
+	req, err := http.NewRequest("POST", o.WarpUrl, bytes.NewBufferString(payload))
+	req.Header.Set("X-Warp10-Token", o.Token)
+	req.Header.Set("Content-Type", "text/plain")
+
+	client := &http.Client{}
```

Comment:
We need to create a single client+transport instead of using a new one or the default each Write.  Check out the http output for an example.